meta/store.go

```diff
@@ -994,11 +1008,22 @@ func (s *Store) Authenticate(username, password string) (ui *UserInfo, err error
 			return ErrUserNotFound
 		}
 
+		// Check the local auth cache first.
+		if p, ok := s.authCache[username]; ok {
```

Comment:
Does `s.read` run this func under a lock?  `s.authCache` needs a mutex around it I think.